### PR TITLE
21 GET Appointment  Available Time

### PR DESCRIPTION
**Pull request by AlaaShurrab**
creating a route that returns all the free time data

### CODE REVIEW

**Review by hassanelnajjar — APPROVED** [review]

File: server/controllers/appointments/getAvailableAppointments/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const { rows: unavailable } = await getUnavailableTimes({ date });
+
+    unavailable.forEach((element) => {
+      const { appointment_time: appointmentTime } = element;
+      if (availableTimeSet.has(appointmentTime)) {
+        availableTimeSet.delete(appointmentTime);
+      }
+    });
+
+    return res.status(201).json({
```
Comment (Mu7ammadAbed) [inline server/controllers/appointments/getAvailableAppointments/index.js:42]:
```suggestion
    return res.json({
```

File: server/controllers/appointments/getAvailableAppointments/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      title: 'available time',
+      detail: 'data collected Successfully',
+      data: Array.from(availableTimeSet),
+    });
+  } catch (error) {
+    return next(error);
+    // return next(boom.badImplementation(error));
```
Comment (Mu7ammadAbed) [inline server/controllers/appointments/getAvailableAppointments/index.js:49]: remove this

File: server/controllers/appointments/getAvailableAppointments/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      '14:00:00',
+      '15:00:00',
+      '16:00:00',
+      '17:00:00',
+      '18:00:00',
+      '19:00:00',
+      '20:00:00',
+      '21:00:00',
+      '22:00:00',
+      '23:00:00',
```
Comment (Mu7ammadAbed) [inline server/controllers/appointments/getAvailableAppointments/index.js:29]: Edit these hours to be 08:00:00 to 18:00:00 and try to edit the appointment hours at the fakeData

File: server/controllers/appointments/getAvailableAppointments/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  try {
+    const { date } = req.params;
+
+    const isValid = await appointmentDateValidation.isValid({
+      appointmentDate: date,
+    });
+
+    if (!isValid) {
+      return next(boomify(400, 'Invalid Date', 'Please send a correct date'));
+      //   return next(boom.conflict('Invalided Date'));
```
Comment (Mu7ammadAbed) [inline server/controllers/appointments/getAvailableAppointments/index.js:14]: remove this

File: server/controllers/appointments/getAvailableAppointments/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      '18:00:00',
+      '19:00:00',
+      '20:00:00',
+      '21:00:00',
+      '22:00:00',
+      '23:00:00',
+    ];
+    const availableTimeSet = new Set(availableTime);
+
+    const { rows: unavailable } = await getUnavailableTimes({ date });
```
Comment (Mu7ammadAbed) [inline server/controllers/appointments/getAvailableAppointments/index.js:33]:
```suggestion
    const { rows: unavailableTimes } = await getUnavailableTimes({ date });
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/controllers/appointments/getAvailableAppointments/index.js (hunk 1 above)
Comment (AlaaShurrab) [reply server/controllers/appointments/getAvailableAppointments/index.js:42]: done

File: server/controllers/appointments/getAvailableAppointments/index.js (hunk 2 above)
Comment (AlaaShurrab) [reply server/controllers/appointments/getAvailableAppointments/index.js:49]: if you mean the comment then done

File: server/controllers/appointments/getAvailableAppointments/index.js (hunk 3 above)
Comment (AlaaShurrab) [reply server/controllers/appointments/getAvailableAppointments/index.js:29]: doen

File: server/controllers/appointments/getAvailableAppointments/index.js (hunk 4 above)
Comment (AlaaShurrab) [reply server/controllers/appointments/getAvailableAppointments/index.js:14]: same as the one above

File: server/controllers/appointments/getAvailableAppointments/index.js (hunk 5 above)
Comment (AlaaShurrab) [reply server/controllers/appointments/getAvailableAppointments/index.js:33]: done

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by hassanelnajjar — APPROVED** [review]

**Pull request merged**